pkg/servicemesh/extension/apply_extension.go

```diff
@@ -148,46 +155,59 @@ func ApplyListenerListPatches(
-	filterMap map[v1alpha1.FilterPhase][]*maistramodel.ExtensionWrapper,
-	phase v1alpha1.FilterPhase) []*hcm_filter.HttpFilter {
+	filterMap map[v1.FilterPhase][]*maistramodel.ExtensionWrapper,
+	phase v1.FilterPhase) []*hcm_filter.HttpFilter {
 	for _, ext := range filterMap[phase] {
-		list = append(list, toEnvoyHTTPFilter(ext))
+		if filter := toEnvoyHTTPFilter(ext); filter != nil {
+			list = append(list, filter)
+		}
 	}
 	filterMap[phase] = []*maistramodel.ExtensionWrapper{}
 	return list
 }
 
 func toEnvoyHTTPFilter(extension *maistramodel.ExtensionWrapper) *hcm_filter.HttpFilter {
+	configuration, err := structpb2.NewStruct(extension.Config.Data)
+	if err != nil {
+		log.Errorf("invalid configuration for extension %s: %v", extension.Name, err)
+		return nil
+	}
+
 	return &hcm_filter.HttpFilter{
 		Name: "envoy.filters.http.wasm",
 		ConfigType: &hcm_filter.HttpFilter_TypedConfig{
-			TypedConfig: util.MessageToAny(&structpb.Struct{
-				Fields: map[string]*structpb.Value{
-					"config": {Kind: &structpb.Value_StructValue{StructValue: &structpb.Struct{Fields: map[string]*structpb.Value{
-						"name":          {Kind: &structpb.Value_StringValue{StringValue: extension.Name}},
-						"rootId":        {Kind: &structpb.Value_StringValue{StringValue: extension.Name + "_root"}},
-						"configuration": {Kind: &structpb.Value_StringValue{StringValue: extension.Config}},
-						"vmConfig": {Kind: &structpb.Value_StructValue{StructValue: &structpb.Struct{Fields: map[string]*structpb.Value{
-							"code": {Kind: &structpb.Value_StructValue{StructValue: &structpb.Struct{Fields: map[string]*structpb.Value{
-								"remote": {Kind: &structpb.Value_StructValue{StructValue: &structpb.Struct{Fields: map[string]*structpb.Value{
-									"httpUri": {Kind: &structpb.Value_StructValue{StructValue: &structpb.Struct{Fields: map[string]*structpb.Value{
-										"uri":     {Kind: &structpb.Value_StringValue{StringValue: extension.FilterURL}},
-										"cluster": {Kind: &structpb.Value_StringValue{StringValue: CacheCluster}},
-										"timeout": {Kind: &structpb.Value_StructValue{StructValue: &structpb.Struct{Fields: map[string]*structpb.Value{
-											"seconds": {Kind: &structpb.Value_NumberValue{NumberValue: 30}},
+			TypedConfig: util.MessageToAny(&udpa.TypedStruct{
+				TypeUrl: "type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm",
+				Value: &structpb.Struct{
+					Fields: map[string]*structpb.Value{
+						"config": {Kind: &structpb.Value_StructValue{StructValue: &structpb.Struct{Fields: map[string]*structpb.Value{
+							"name":   {Kind: &structpb.Value_StringValue{StringValue: extension.Name}},
+							"rootId": {Kind: &structpb.Value_StringValue{StringValue: extension.Name + "_root"}},
+							"configuration": {Kind: &structpb.Value_StructValue{StructValue: &structpb.Struct{Fields: map[string]*structpb.Value{
+								"@type": {Kind: &structpb.Value_StringValue{StringValue: "type.googleapis.com/google.protobuf.Struct"}},
```

Comment:
does this actually work? I thought we'd have to pass a `string`

Reply:
Works like a charm :)

Reply:
awesome, so we can even hand off the marshaling to proto